code/game/turfs/space/space.dm

```diff
@@ -57,39 +56,24 @@
 			ReplaceWithLattice()
 		return
 
-	if (istype(C, /obj/item/stack/tile/floor))
-		var/obj/structure/lattice/L = locate(/obj/structure/lattice, src)
-		if(L)
-			var/obj/item/stack/tile/floor/S = C
-			if (S.get_amount() < 1)
-				return
-			qdel(L)
-			playsound(src, 'sound/weapons/Genhit.ogg', 50, 1)
-			S.use(1)
-			ChangeTurf(/turf/simulated/floor/airless)
-			return
-		else
-			to_chat(user, SPAN_WARNING("The plating is going to need some support."))
-			return
 	if (istype(C, /obj/item/stack/material))
 		var/obj/item/stack/material/M = C
 		var/material/mat = M.get_material()
 		if (!mat.name == MATERIAL_STEEL)
 			return
+
 		var/obj/structure/lattice/L = locate(/obj/structure/lattice, src)
 		if(L)
-			var/obj/item/stack/tile/S = C
-			if (S.get_amount() < 1)
-				return
-			qdel(L)
+			var/obj/item/stack/tile/floor/S = C
```

Comment:
Why do you cast C as a `obj/item/stack/tile/floor` when it's actually a `/obj/item/stack/material` (as checked before in the if)

Reply:
Suspicious.

Reply:
I didn't really thought about this part when adjusting the parts, but I believe this is a remnant of ability to use tiles on lattices.
Let me remove it in a moment.

Reply:
I edited out those lines.
It's really odd that this code worked.